api/server/server.go

```diff
@@ -43,15 +43,22 @@ type Server struct {
 	Datastore    models.Datastore
 }
 
-func New(ctx context.Context, ds models.Datastore, mq models.MessageQueue, r *runner.Runner, tasks chan task.Request, enqueue models.Enqueue) *Server {
+type Components struct {
```

Comment:
I am very skeptical of this change. We are just instituting that Server has too many responsibilities: operate datastore, message queueing, container running and etc. 

And worse, this is not really handling the case in which `Components` is passed on incomplete. 

This seems to me a rather gratuitous change - that in attempting to solve an aesthetic issue, it is actually adding more issues, and more importantly, it is not solving the underlying problem of concentration of responsibilities.

I think we might be barking to the wrong tree here.

What do you think of actually making package `server` smaller? We could use concurrent programming to compose this package better, IMO. 

For the sake of exercise, I would try to refactor `server` with just a thin machinery comprising: a channel for job enqueuing and datastore.

Out of this job enqueuing job, either direct the load to a `*Runner` (also connected through a channel), or to the `MessageQueue` (perhaps also connected through a channel).

With this approach, `Server` will become significantly smaller - will have fewer dependencies, and this function signature problem will be solved as a side effect.